### ISSUE TITLE
Plugin missing element dependencies

### ISSUE THREAD

**Issue by nm17** [DeflatedPickle/Quiver#18]
Windows 10, Oracle java 8, https://github.com/DeflatedPickle/Quiver/releases/tag/v1.0.0-alpha

Bat file logs:
```
[INFO ] 00:46:25 (636 ms since JVM launch) [main/1] com.deflatedpickle.quiver.launcher.MainKt: Installed JANSI for this session
        com.deflatedpickle.quiver.launcher.MainKt.main(main.kt:41)
[INFO ] 00:46:26 (1064 ms since JVM launch) [main/1] com.deflatedpickle.quiver.launcher.MainKt: Registered a default exception handler
        com.deflatedpickle.quiver.launcher.MainKt.main(main.kt:82)
[INFO ] 00:46:26 (1095 ms since JVM launch) [main/1] com.deflatedpickle.quiver.launcher.MainKt: Created the plugins folder at C:\Users\nm17\Documents\Quiver-1.0.0\plugins
        com.deflatedpickle.quiver.launcher.MainKt.main(main.kt:89)
[DEBUG] 00:46:26 (1095 ms since JVM launch) [main/1] EventCreateFile: This event was triggered with class java.io.File
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1095 ms since JVM launch) [main/1] EventCreateFile: This event was triggered with class java.io.File
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[INFO ] 00:46:26 (1102 ms since JVM launch) [main/1] ClassGraphUtil: Found the plugin JARs: [core.jar, filepanel.jar, filetable.jar, foldertree.jar, imageviewer.jar, launcher.jar, tableviewer.jar, textviewer.jar, treeviewer.jar]
        com.deflatedpickle.haruhi.util.ClassGraphUtil.<clinit>(ClassGraphUtil.kt:29)
[DEBUG] 00:46:26 (1202 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin ImageViewerPlugin from com.deflatedpickle.imageviewer
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1202 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1202 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin Quiver from com.deflatedpickle.quiver
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1202 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1202 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin FilePanel from com.deflatedpickle.quiver.filepanel
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin FileTable from com.deflatedpickle.quiver.filetable
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin FolderTree from com.deflatedpickle.quiver.foldertree
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin Launcher from com.deflatedpickle.quiver.launcher
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1217 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin TextViewerPlugin from com.deflatedpickle.quiver.textviewer
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1233 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1233 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin TableViewerPlugin from com.deflatedpickle.tableviewer
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1233 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[DEBUG] 00:46:26 (1233 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found the plugin TreeViewerPlugin from com.deflatedpickle.treeviewer
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:93)
[DEBUG] 00:46:26 (1233 ms since JVM launch) [main/1] EventDiscoverPlugin: This event was triggered with class com.sun.proxy.$Proxy24
        com.deflatedpickle.haruhi.api.event.AbstractEvent.trigger(AbstractEvent.kt:15)
[INFO ] 00:46:26 (1233 ms since JVM launch) [main/1] com.deflatedpickle.haruhi.util.PluginUtil: Found 9 plugin/s
        com.deflatedpickle.haruhi.util.PluginUtil.discoverPlugins(PluginUtil.kt:112)
[DEBUG] 00:46:26 (1233 ms since JVM launch) [main/1] com.deflatedpickle.quiver.launcher.MainKt: Validated all plugins with 0 error/s
        com.deflatedpickle.quiver.launcher.MainKt.main(main.kt:111)
[WARN ] 00:46:26 (1233 ms since JVM launch) [main/1] com.deflatedpickle.quiver.launcher.MainKt: main threw java.lang.annotation.IncompleteAnnotationException: com.deflatedpickle.haruhi.api.plugin.Plugin missing element dependencies
        com.deflatedpickle.quiver.launcher.MainKt$main$2.uncaughtException(main.kt:71)
```

Pop up menu:
```
java.lang.annotation.IncompleteAnnotationException: com.deflatedpickle.haruhi.api.plugin.Plugin missing element dependencies
	at io.github.classgraph.AnnotationInfo$AnnotationInvocationHandler.invoke(AnnotationInfo.java:418)
	at com.sun.proxy.$Proxy24.dependencies(Unknown Source)
	at com.deflatedpickle.haruhi.api.plugin.DependencyComparator.compare(DependencyComparator.kt:8)
	at com.deflatedpickle.haruhi.api.plugin.DependencyComparator.compare(DependencyComparator.kt:6)
	at java.base/java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:216)
	at java.base/java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:216)
	at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.base/java.util.TimSort.sort(TimSort.java:220)
	at java.base/java.util.Arrays.sort(Arrays.java:1515)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1750)
	at java.base/java.util.Collections.sort(Collections.java:179)
	at kotlin.collections.CollectionsKt__MutableCollectionsJVMKt.sortWith(MutableCollectionsJVM.kt:42)
	at com.deflatedpickle.quiver.launcher.MainKt.main(main.kt:114)
```

**Comment by DeflatedPickle** [comment]
So, if this isn't caused by #17, I'm completely stumped on it. I've never seen this kind of error and will have to look into it. Though you will possibly run into #15 if it is fixed by my suggestion in #17, until I migrate some things. I was a bit busy today, sorry!

**Comment by nm17** [comment]
Just in case, I tried to build distZip and run the bat file (HEAD 7cdfb7c4ca36079e23a915098097a0952d612412) and the same error happens.
![explorer_TW420ZEfCV](https://user-images.githubusercontent.com/23419131/96285658-4b9bc700-0ff0-11eb-96d4-cdb96a07a5f2.png)

**Comment by DeflatedPickle** [comment]
This is an incomplete error. I can't help unless you expand that "more details" bit

**Comment by nm17** [comment]
@DeflatedPickle it's the same as the last one.

> Pop up menu:
> 
> ```
> java.lang.annotation.IncompleteAnnotationException: com.deflatedpickle.haruhi.api.plugin.Plugin missing element dependencies
> 	at io.github.classgraph.AnnotationInfo$AnnotationInvocationHandler.invoke(AnnotationInfo.java:418)
> 	at com.sun.proxy.$Proxy24.dependencies(Unknown Source)
> 	at com.deflatedpickle.haruhi.api.plugin.DependencyComparator.compare(DependencyComparator.kt:8)
> 	at com.deflatedpickle.haruhi.api.plugin.DependencyComparator.compare(DependencyComparator.kt:6)
> 	at java.base/java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:216)
> 	at java.base/java.util.Comparator.lambda$thenComparing$36697e65$1(Comparator.java:216)
> 	at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
> 	at java.base/java.util.TimSort.sort(TimSort.java:220)
> 	at java.base/java.util.Arrays.sort(Arrays.java:1515)
> 	at java.base/java.util.ArrayList.sort(ArrayList.java:1750)
> 	at java.base/java.util.Collections.sort(Collections.java:179)
> 	at kotlin.collections.CollectionsKt__MutableCollectionsJVMKt.sortWith(MutableCollectionsJVM.kt:42)
> 	at com.deflatedpickle.quiver.launcher.MainKt.main(main.kt:114)
> ```

**Comment by DeflatedPickle** [comment]
Apparently this can be caused by a higher Java version than Java 8. Are you *sure* you're running *with* Java 8? I'll have the logger log the running Java version so you can check

**Comment by nm17** [comment]
@DeflatedPickle I uninstalled all java versions and installed only `openjdk8` and `openjdk8jre` from chocolatey. Now it gives me the error from #15. I think you can close this issue for now (maybe add a note about java 8 in README for new users?).

**Comment by nm17** [comment]
Also dropping support for Java 8 later on might be a good idea, but that's just my thought and not needed right now.

**Comment by DeflatedPickle** [comment]
I haven't had time to update the README as I've been trying to fix #15, but I will. The problem with updating is most people who've tried this have used Java 8, which means that's the latest they have
